src/components/QrScan/QrScan.js

```diff
@@ -67,20 +67,21 @@ const QrScan = () => {
       }
       return;
     }
-
     qrScan = new QrScanner(
       videoElement,
-      (data) => {
-        setScannedData(data);
+      (results) => {
+        setScannedData(results.data);
       },
-      () => {},
-      () => ({
-        // define scan region for QrScanner
-        x: 0,
-        y: 0,
-        width: videoElement.videoWidth,
-        height: videoElement.videoHeight,
-      })
+      {
+        // eslint-disable-next-line no-shadow
```

Comment:
I think you could create a new useRef for the running list of scanned codes, rather than useState, and then remove the item from the useEffect dependency array.